crates/json-abi/src/abi.rs

```diff
@@ -246,6 +253,40 @@ impl JsonAbi {
         }
     }
 
+    /// Goes over all structs referenced as internal types in the ABI and changes them to global
+    /// structs.
+    ///
+    /// Namespace for global struct names is managed and in cases of duplicates structs are renamed.
+    pub fn rename_internal_structs(&mut self) {
+        let mut collector = Collector::default();
+        self.walk(&mut collector);
+
+        let Collector { unique_struct_components } = collector;
+
+        // Collect needed renames.
+        let renames = unique_struct_components
+            .into_iter()
+            .filter(|(_, components)| components.len() > 1)
+            .map(|(name, components)| {
+                (
+                    name.to_string(),
+                    components
+                        .into_iter()
+                        .enumerate()
+                        .map(move |(idx, components)| {
+                            let components =
+                                components.into_iter().map(ToString::to_string).collect::<Vec<_>>();
+                            (components, format!("{name}{idx}"))
```

Comment:
should we start at 0 here or at 1?

Reply:
function overloads start with 0, so I've just made it similar

just noticed that functions are named as `function_0`, `function_1`, should we include underscore here as well?